Modules/IO/MINC/src/itkMINCImageIO.cxx

```diff
@@ -497,13 +505,29 @@ MINCImageIO::ReadImageInformation()
       sep[i - 1] = _sep;
 
       this->SetDimensions(i - 1, static_cast<unsigned int>(_sz));
-      this->SetDirection(i - 1, _dir);
       this->SetSpacing(i - 1, _sep);
 
       ++usable_dimensions;
     }
   }
 
+  // Transform MINC PositiveCoordinateOrientation RAS coordinates to
+  // internal ITK PositiveCoordinateOrientation LPS Coordinates
+  dir_cos = RAS_tofrom_LPS * dir_cos;
+
+  // Transform origin coordinates
+  o_origin = dir_cos * origin;
+
+  for (int i = 0; i < spatial_dimension_count; ++i)
```

Comment:
If `spatial_dimension_count` is an unsigned int this will trigger a compiler warning: use `unsigned int` for `i` (or even better, a range-based loop with `auto`) if `spatial_dimension_count` is an `unsigned int`.

Reply:
Change spatial_dimension_count to unsigned int.